x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/tree/EsqlNodeSubclassTests.java

```diff
@@ -142,8 +142,8 @@ Expression randomResolvedExpression(Class<?> argClass) throws Exception {
         Class<? extends Expression> asNodeSubclass = (Class<? extends Expression>) argClass;
         if (Modifier.isAbstract(argClass.getModifiers())) {
             while (true) {
-                var candidate = randomFrom(subclassesOf(asNodeSubclass));
-                if (UNRESOLVED_CLASSES.contains(candidate) == false) {
+                var candidate = randomFrom(subclassesOf(asNodeSubclass, CLASSNAME_FILTER));
+                if (UNRESOLVED_CLASSES.stream().allMatch(unresolved -> unresolved.isAssignableFrom(candidate) == false)) {
```

Comment:
Good catch!

> I'm confused why this didn't bite us earlier.

probably just because the only missing case was `UnresolvedNamePattern`, that is a recent addition
At this point I think we can remove `UnresolvedAlias` and `UnresolvedStar` from the list, that are subclasses of `UnresolvedNamedExpression`